js/app.js

```diff
@@ -1,262 +1,269 @@
-function AppViewModel() {
-  const self = this;
-  self.requestMethod = ko.observable();
-  self.requestUrl = ko.observable();
-  self.responseBody = ko.observable();
-  self.callDuration = ko.observable();
-  self.callStatus = ko.observable();
-  self.responseHeaders = ko.observableArray();
-  self.requestHeaders = ko.observableArray();
-
-  self.showRequestHeaders = ko.observable(true);
-  self.showRequestBody = ko.observable(false);
-
-  self.showResponseHeaders = ko.observable(false);
-  self.showResponseBody = ko.observable(true);
-  
-  self.useFormattedResponseBody = ko.observable(false);
-  self.useRawResponseBody = ko.observable(true);
-  
-  self.bodyType = ko.observable();
-  self.formDataParams = ko.observableArray();
-  self.formEncodedParams = ko.observableArray();
-  self.rawBody = ko.observable();
-  
-  self.bookmarks = ko.observableArray();
-  
- 
-  
-  localforage.config({
-    name: 'resting',
-    storeName: 'bookmarks'
-});
-  
-  
-  let convertToFormData = function(data) {
-    let formDataObj = {};
-    if(data) {
-      data.forEach(function(entry) {
-        formDataObj[entry.name] = entry.value;
-      });
-    }
-    return formDataObj;
-  };
-  
-  let loadBookmarks = function() {
-    let bookmarks = [];
-    localforage.iterate(function(value,key, iterationNumber) {
-      self.bookmarks.push(JSON.parse(value));
+(($, localforage, ko) => {
+  const makeBookmark = (id, request, name) => ({ id, request, name });
+
+  const makeRequest = (method, url, headers, bodyType, body) =>
+    ({ method, url, headers, bodyType, body });
+
+  function AppViewModel() {
+    const Resting = {
```

Comment:
my curiosity: what is the idea behind collecting all view fields ins object ?

Reply:
There are several reasons.

The first is the issue with _this_ in Javascript. As you can change the scope of any object any time, you could have attributes of method rewritten.

Such issues are very complicated to debug.

And that is connected to a design concern: encapsulation.

If you create an object to return with all the accessible fields and method, you can hide the stuff you don't want some developer access or change.

It is called "Revealed Module Pattern", here you can find a better documentation: https://scotch.io/bar-talk/4-javascript-design-patterns-you-should-know